test/extended/olm/olmv1.go

```diff
@@ -99,6 +103,83 @@ var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLM][Skipped:Disconnected] OLM
 	})
 })
 
+var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLM][Skipped:Disconnected] OLMv1 Catalogs /v1/api/all endpoint", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("default")
+
+	g.It("should serve FBC", func(ctx g.SpecContext) {
+		checkFeatureCapability(ctx, oc)
+
+		g.By("Testing /api/v1/all endpoint for catalog openshift-community-operators")
+		verifyAPIEndpoint(ctx, oc, oc.Namespace(), "openshift-community-operators", "all")
+	})
+})
+
+var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLMCatalogdAPIV1Metas][Skipped:Disconnected] OLMv1 Catalogs /v1/api/metas endpoint", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("default")
+	g.It(" should serve the /v1/api/metas API endpoint", func(ctx g.SpecContext) {
+		checkFeatureCapability(ctx, oc)
+
+		g.By("Testing api/v1/metas endpoint for catalog openshift-community-operators")
+		verifyAPIEndpoint(ctx, oc, oc.Namespace(), "openshift-community-operators", "metas?schema=olm.package")
+	})
+})
+
+var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLM][Skipped:Disconnected] OLMv1 Catalogs /v1/api/all endpoint load test", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("default")
+	providedCatalogs, idx := []string{
+		"openshift-certified-operators",
+		"openshift-community-operators",
+		"openshift-redhat-marketplace",
+		"openshift-redhat-operators",
+	}, 0
+	g.It("should be able to access /v1/api/all API endpoints of all catalogs within a reasonable amount of time", func(ctx g.SpecContext) {
+		checkFeatureCapability(ctx, oc)
+
+		var wg sync.WaitGroup
+		for range 100 {
+			wg.Add(1)
+			go func(catalogIdx int) {
+				defer wg.Done()
+
+				g.By(fmt.Sprintf("Testing api/v1/all endpoint for catalog %s", providedCatalogs[catalogIdx]))
+				verifyAPIEndpoint(ctx, oc, oc.Namespace(), providedCatalogs[catalogIdx], "all")
+			}(idx)
+			idx = (idx + 1) % len(providedCatalogs)
+		}
```

Comment:
I'm having a _really_ hard time grokking the way the index is working here. For clarity, can we change this to something like:

```go
var wg sync.WaitGroup
for _, catalog := range providedCatalogs {
    for range 100 {
        wg.Add(1)
        go func() {
            verifyAPIEndpoint(ctx, oc, oc.Namespace(), catalog, "all")
        }()
    }
}
wg.Wait()
```

Reply:
Also, I don't think you need separate Jobs. Might take some experimentation, but you may be able to set 
```yaml
spec:
    activeDeadlineSeconds: 30  # how long the job is allowed to be active
    backoffLimit: 0 # don't retry failed pods
    completions: 100 # required 100 completions
    maxFailedIndices: 0 # allow 0 failures
    parallelism: 100 # run 100 pods at a time
    ttlSecondsAfterFinished: 10 # make the job eligible for deletion 10 seconds after it finishes
```